### DISCUSSION TITLE
2023/03/10/%E6%97%A0%E7%9F%A5%E6%97%B6%E8%AF%8B%E6%AF%81%E5%8E%9F%E7%A5%9E/

### DISCUSSION THREAD

**Discussion by giscus** [xjcwliucy/xjcwliucy.github.io#1] [Show and tell]
# 2023/03/10/%E6%97%A0%E7%9F%A5%E6%97%B6%E8%AF%8B%E6%AF%81%E5%8E%9F%E7%A5%9E/

Solution:  题意: 原题面已经给得很清楚——题目传送门 显然，这是一道数论题。   做法： 对于 nnn 为偶数：输出 -1。 对于 nnn 为奇数： 构造 $F_{(2i\bmod n)}=i (i\in [0,n-1]\cap \mathbb{N}) $，两次遍历 CiC_iCi​ 并输出 FCiF_{C_i}FCi​​。   分析&做法解释：   首先不难发现，只要数据有

http://localhost:4000/2023/03/10/%E6%97%A0%E7%9F%A5%E6%97%B6%E8%AF%8B%E6%AF%81%E5%8E%9F%E7%A5%9E/

**Comment by xjcwliucy** [comment]
1.0.4中对 $n$ 为偶数的不可行性证明并不严谨，然而我也想不出更正确的证明。但一个有说服力的理由是，这份代码是正解，至少是AC代码，所以猜想一定正确（至少在这道题的数据范围内）。

**Comment by xjcwliucy** [comment]
Although the proof in 1.0.4 when $n\equiv 0(\bmod 2)$ are not so correct,I don't know the proof more correct.But a compelling reason is that this code has been accepted,So the guess must be correct (at least within the data range of this question)
